File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let res = nums[0];
    let maxEnding = nums[0];
    
    for (let i = 1; i < nums.length; i++) {
        let v1 = maxEnding + nums[i];
        let v2 = nums[i];
        maxEnding = Math.max(v1, v2);
        res = Math.max(res, maxEnding);
    }
    return res;
};

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int res = nums[0];
        int maxEnding = nums[0];

        for (int i = 1; i < nums.size(); i++){
            int v1 = maxEnding + nums[i];
            int v2 = nums[i];
            maxEnding = max(v1, v2);
            res = max(res, maxEnding);
        }
        return res;
    }
};